File: py/IEC62325/InfIEC62325/InfEnergyScheduling/DynamicSchedule.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106 on Wed Dec 27 12:50:20 2023

from typing import Optional

from IEC61970.Base.Core.BasicIntervalSchedule import BasicIntervalSchedule
from IEC62325.MarketOperations.MarketOpCommon.MktMeasurement import MktMeasurement
from IEC62325.MarketOperations.ReferenceData.SubControlArea import SubControlArea


class DynamicSchedule(BasicIntervalSchedule):

    def __init__(self) -> None:
        super().__init__()
        self.dyn_sched_sign_revOptional[bool] = False  # Dynamic schedule sign reversal required (true/false)
        self.dyn_sched_statusOptional[str] = ""  # The "active" or "inactive" status of the dynamic schedule
        self.send_sub_control_areaOptional[SubControlArea] = SubControlArea()  # A control area can send dynamic schedules to other control areas
        self.receive_sub_control_areaOptional[SubControlArea] = SubControlArea()  # A control area can receive dynamic schedules from other control areas
        self.mkt_measurementOptional[MktMeasurement] = MktMeasurement()

File: java/IEC62325/InfIEC62325/InfEnergyScheduling/DynamicSchedule.java
package IEC62325.InfIEC62325.InfEnergyScheduling;

import IEC61970.Base.Domain.Boolean;
import IEC61970.Base.Domain.String;
import IEC61970.Base.Core.BasicIntervalSchedule;
import IEC62325.MarketOperations.ReferenceData.SubControlArea;
import IEC62325.MarketOperations.MarketOpCommon.MktMeasurement;

/**
 * A continuously variable component of a control area's MW net interchange
 * schedule. Dynamic schedules are sent and received by control areas.
 * @created 03-Jan-2024 1:49:41 PM
 */
public class DynamicSchedule extends BasicIntervalSchedule {

	/**
	 * Dynamic schedule sign reversal required (true/false)
	 */
	public Boolean dynSchedSignRev;
	/**
	 * The "active" or "inactive" status of the dynamic schedule
	 */
	public String dynSchedStatus;
	/**
	 * A control area can send dynamic schedules to other control areas
	 */
	public SubControlArea Send_SubControlArea;
	/**
	 * A control area can receive dynamic schedules from other control areas
	 */
	public SubControlArea Receive_SubControlArea;
	public MktMeasurement MktMeasurement;

	public DynamicSchedule(){

	}
}//end DynamicSchedule